### PR TITLE
Fix some documentation in conch

### PR DESCRIPTION
**Pull request by mmattice**
https://twistedmatrix.com/trac/ticket/9689

## Contributor Checklist:

* [x] There is an associated ticket in Trac. Create a new one at https://twistedmatrix.com/trac/newticket
* [x] The changes pass minimal style checks (see: https://twistedmatrix.com/trac/wiki/TwistedDevelopment#GettingYourPatchAccepted )
* [x] I have created a newsfragment in src/twisted/newsfragments/ (see: https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles )
* [x] I have updated the automated tests.

### CODE REVIEW

File: src/twisted/conch/interfaces.py [hunk 1]
```diff
@@ -363,8 +364,8 @@ def matchesHost(hostname):
         address, you have to resolve it yourself, and pass it in as a dotted
         quad string.
 
-        @param key: The hostname to match against.
-        @type key: L{str}
+        @param hostname: The hostname to match against.
+        @type hostname: L{str}
```
Comment (twm) [inline src/twisted/conch/interfaces.py]: It looks like this is `L{str} or L{bytes}` in the `PlainHosts` and `HashedHosts` implementations (though their docs claim just `L{bytes}`). I filed https://twistedmatrix.com/trac/ticket/9713 for this.

File: src/twisted/conch/interfaces.py [hunk 2]
```diff
@@ -150,8 +150,9 @@ def openFile(filename, flags, attrs):
 
         @param filename: a string representing the file to open.
 
-        @param flags: an integer of the flags to open the file with, ORed together.
-        The flags and their values are listed at the bottom of this file.
+        @param flags: an integer of the flags to open the file with, ORed
+        together.  The flags and their values are listed at the bottom of
+        L{twisted.conch.ssh.filetransfer} as FXF_*.
```
Comment (twm) [inline src/twisted/conch/interfaces.py:155]: Unfortunately these don't seem to show up in our API docs: https://twistedmatrix.com/documents/current/api/twisted.conch.ssh.filetransfer.html

image

Still, the "view source" link gets you to the right place, so this is a useful improvement. I filed https://twistedmatrix.com/trac/ticket/9714 to properly document these constants.

File: src/twisted/conch/interfaces.py (hunk 1 above)
Comment (twm) [inline src/twisted/conch/interfaces.py:368]: It looks like this is `L{str} or L{bytes}` in the `PlainHosts` and `HashedHosts` implementations (though their docs claim just `L{bytes}`). I filed https://twistedmatrix.com/trac/ticket/9713 for this.

**Review by twm — APPROVED** [review]
Hi @mmattice,

I noticed a few more things that could be improved, but this PR is a definite improvement on what's there so I'm going to go ahead and merge. If either of the tickets I filed interest you feel free. :)

It looks like @glyph added a [newsfragment](https://twistedmatrix.com/trac/wiki/ReviewProcess#Newsfiles) for you. Next time it would be helpful to include one. For small documentation changes like this an empty `<ticket number>.misc` file is fine.

Thanks for contributing to Twisted!

File: src/twisted/conch/interfaces.py (hunk 2 above)
Comment (glyph) [reply src/twisted/conch/interfaces.py:155]: `@var` annotations at the module level is what it takes to get these to show up.

**Pull request merged**